### PR TITLE
Faster, lock-free pool

### PR DESCRIPTION
**Pull request by dim**
Before:
```
$ go test ./... -run=NONE -bench=Pool -benchmem -cpu=1,2,4,8 -benchtime=2s
BenchmarkPool   10000000         293 ns/op         0 B/op        0 allocs/op
BenchmarkPool-2  3000000        1127 ns/op         0 B/op        0 allocs/op
BenchmarkPool-4  2000000        1187 ns/op         0 B/op        0 allocs/op
BenchmarkPool-8  2000000        1459 ns/op         0 B/op        0 allocs/op
```

After:
```
$ go test ./... -run=NONE -bench=Pool -benchmem -cpu=1,2,4,8 -benchtime=2s
BenchmarkPool   10000000         330 ns/op         0 B/op        0 allocs/op
BenchmarkPool-2 10000000         288 ns/op         0 B/op        0 allocs/op
BenchmarkPool-4 10000000         401 ns/op         0 B/op        0 allocs/op
BenchmarkPool-8 10000000         403 ns/op         0 B/op        0 allocs/op
```

### CODE REVIEW

File: pool.go [hunk 1]
```diff
@@ -195,92 +197,67 @@ func (p *connPool) Put(cn *conn) error {
...
-	p.closed = true
 	p.rl.Close()
-	var retErr error
+
 	for {
-		e := p.conns.Front()
-		if e == nil {
-			break
+		if p.Size() < 1 {
+			return
```
Comment (vmihailenco) [inline pool.go]: You couild replace `return` with `break` and get rid of `panic`

File: redis.go [hunk 2]
```diff
@@ -167,6 +168,7 @@ type Options struct {
 	WriteTimeout time.Duration
 
 	PoolSize    int
+	PoolTimeout time.Duration
```
Comment (vmihailenco) [inline redis.go]: I know that I am bad at documenting my code, but could you please add brief comment explaining this option? Something like `maximum duration before timing out waiting for connection from the pool`.

**Comment by dim** [comment]
Updated

**Comment by vmihailenco** [comment]
Thanks! Finally package got some docs :)

**Pull request merged**